internal/apis/certmanager/validation/certificate.go

```diff
@@ -47,12 +48,46 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 
 	el = append(el, validateIssuerRef(crt.IssuerRef, fldPath)...)
 
-	if len(crt.CommonName) == 0 && len(crt.DNSNames) == 0 && len(crt.URISANs) == 0 && len(crt.EmailSANs) == 0 && len(crt.IPAddresses) == 0 {
+	var commonName = crt.CommonName
+	if crt.LiteralSubject != "" {
+
+		if !utilfeature.DefaultFeatureGate.Enabled(feature.LiteralCertificateSubject) {
+			el = append(el, field.Forbidden(fldPath.Child("literalSubject"), "Feature gate LiteralCertificateSubject must be enabled"))
+		}
+
+		sequence, err := pki.ParseSubjectStringToRdnSequence(crt.LiteralSubject)
+		if err != nil {
+			el = append(el, field.Invalid(fldPath.Child("literalSubject"), crt.LiteralSubject, err.Error()))
+		}
```

Comment:
> I'm not sure whether that is the Name validation you are referring to

I just meant not allowing random keys- I see that this now does what that 👍🏼 

Very minor nit: would be good if the error message printed the unrecognized key.

Reply:
I wanted to print the key, however, it would basically mean re-implementing the parser or implement some heuristic. The parsed result (RDNSequence) doesn't have the original string attached to it. So then I would either do something with splitting the string on comma and + and use the index to find which item I should take. Or lookup `=$value` in the string and take the preceding key. It can be done obviously, but is it worth it?